src/fsharp/ReferenceResolution.fs

```diff
@@ -106,7 +112,10 @@ module internal MSBuildResolver =
             | Net35 ->  Some TargetDotNetFrameworkVersion.Version35
             | Net40 ->  Some TargetDotNetFrameworkVersion.Version40
             | Net45 ->  Some TargetDotNetFrameworkVersion.Version45
+#if BUILDING_PROTO
+#else
             | Net451 -> Some TargetDotNetFrameworkVersion.Version451
```

Comment:
The current OSS build incorrectly sets the FX_ATLEAST_45 variable, if we ensure the build doesn't set this variable then we don't need these additional #defines.  However now we have warnings as errors we will need to ensure the unused parameters don't cause the warnings.